crates/shared/src/sources/balancer_v2/pool_storage.rs

```diff
@@ -197,6 +202,44 @@ where
     }
 }
 
+#[async_trait::async_trait]
+impl<Factory> EventStoring<BasePoolFactoryEvent> for PoolStorage<Factory>
+where
+    Factory: FactoryIndexing,
+{
+    async fn replace_events(
+        &mut self,
+        events: Vec<Event<BasePoolFactoryEvent>>,
+        range: RangeInclusive<BlockNumber>,
+    ) -> Result<()> {
+        tracing::debug!("replacing {} events for block {:?}", events.len(), range);
+
+        self.remove_pools_newer_than_block(range.start().to_u64());
+        self.append_events(events).await
+    }
+
+    async fn append_events(&mut self, events: Vec<Event<BasePoolFactoryEvent>>) -> Result<()> {
```

Comment:
If were no longer have an event handle what are all these events handling method still used for?

Reply:
We do. Its in the `pool_storage` and `pool_fetching::registry` modules where we now have a registry per factory instead of a BalancerPoolRegistry for all three (which required manually adding a new field and code for each new factory you wanted to index).